### PR TITLE
Bump to v5.0.2

### PR DESCRIPTION
**Pull request by mheon**
As the title says

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
I thought we did a "-dev" version only on main branches.  That way when we need to create a patch for RHEL, it doesn't get the -dev tag later on down the road.

Otherwise, LGTM

**Comment by mheon** [comment]
This is not a RHEL branch. We only omit -dev on RHEL branches

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Release Notes
 
+## 5.0.2
+### Bugfixes
+- Fixed a bug that could leak IPAM entries when a network was removed ([#22034](https://github.com/containers/podman/issues/22034)).
+- Fixed a bug that could cause the rootless network namespace to not be cleaned up on if an error occurred during setupm resulting in errors relating to a missing resolv.conf being displayed ([#22168](https://github.com/containers/podman/issues/22168)).
```
Comment (Luap99) [inline RELEASE_NOTES.md:6]:
```suggestion
- Fixed a bug that could cause the rootless network namespace to not be cleaned up on if an error occurred during setup resulting in errors relating to a missing resolv.conf being displayed ([#22168](https://github.com/containers/podman/issues/22168)).
```

**Review by Luap99 — COMMENTED** [review]
LGTM besides one typo, although not blocking for me

File: RELEASE_NOTES.md (hunk 1 above)
Comment (mheon) [reply RELEASE_NOTES.md]: Fixed

**Pull request merged**